### PR TITLE
some fixes for ruff code C41 (about loops and iteration)

### PR DESCRIPTION
**Pull request by fchapoton**
a few minor code fixes in various places, about a better use of loops

https://docs.astral.sh/ruff/rules/#flake8-comprehensions-c4

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/geometry/hyperplane_arrangement/arrangement.py [hunk 1]
```diff
@@ -1887,17 +1887,17 @@ def _parallel_hyperplanes(self):
...
             A = through_origin.A()
             b = hyperplane.b() * (A / hyperplane.A())
             parallel_planes.append([b, (hyperplane, A, b)])
             parallels[through_origin] = parallel_planes
-        parallels = [tuple(tuple(hyperplane[1]
-                           for hyperplane in sorted(parallels[key])))
-                     for key in parallels.keys()]
+        parallels = [tuple(hyperplane[1]
+                           for hyperplane in sorted(parallels[key]))
+                     for key in parallels]
```
Comment (mkoeppe) [inline src/sage/geometry/hyperplane_arrangement/arrangement.py]: better to rewrite using `parallels.items()`

File: src/sage/matroids/matroids_plot_helpers.py [hunk 2]
```diff
@@ -172,10 +172,11 @@ def it(M, B1, nB1, lps):
             cc = interval*j
             pts[i[j-1]] = (cc*pt1[0]+(1-cc)*pt2[0], cc*pt1[1]+(1-cc)*pt2[1])
     trilines = [list(set(x)) for x in lines if len(x) >= 3]
-    curvedlines = [list(set(list(x)).difference(set(lps)))
-                   for x in M.flats(2) if set(list(x)) not in trilines if
-                   len(list(x)) >= 3]
-    nontripts = [i for i in nB1 if i not in pts.keys()]
+    set_lps = set(lps)
+    curvedlines = [list(sx.difference(set_lps))
+                   for x in M.flats(2) if (sx := set(x)) not in trilines
```
Comment (mkoeppe) [inline src/sage/matroids/matroids_plot_helpers.py:174]: is this condition ever true? `trilines` appears to be a list of lists

**Review by mkoeppe — APPROVED** [review]
Otherwise LGTM

File: src/sage/matroids/matroids_plot_helpers.py (hunk 2 above)
Comment (fchapoton) [reply src/sage/matroids/matroids_plot_helpers.py:174]: This part of the code looked strange before my changes. I am not sure what is intended by the author.

Comment (mkoeppe) [reply src/sage/matroids/matroids_plot_helpers.py:174]: OK.

**Comment by vbraun** [comment]
merge conflict

**Review by mkoeppe — APPROVED** [review]
LGTM

**Pull request merged**